### ISSUE TITLE
installation issue on windows

### ISSUE THREAD

**Issue by williamcioffi** [r-glennie/CTMCdive#15]
Versions:

```
R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)
```

Error:

```
: ctmc_dive.o: too many sections (45373)
: Assembler messages:
: Fatal error: can't write 315 bytes to section .text of ctmc_dive.o: 'file too big'
: Fatal error: ctmc_dive.o: file too big
```

Possible references:
https://github.com/google/googletest/issues/1841
https://stackoverflow.com/questions/16596876/gcc-equivalent-of-mss-bigobj

**Comment by dill** [comment]
I wasn't able to duplicate this on my own Windows 10 machine. Can you let me know the gcc/clang version being used?

**Comment by williamcioffi** [comment]
The call that fails:
```
g++ -std=gnu++11  -I"C:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG -I"C:/Users/user/AppData/Local/R/WIN-LI~1/4.2/TMB/include" -I"C:/Users/user/AppData/Local/R/WIN-LI~1/4.2/RCPPEI~1/include"   -DTMB_EIGEN_DISABLE_WARNINGS -DLIB_UNLOAD=R_unload_ctmc_dive  -DTMB_LIB_INIT=R_init_ctmc_dive  -DCPPAD_FRAMEWORK -I'C:/Users/user/AppData/Local/R/win-library/4.2/TMB/include' -I'C:/Users/user/AppData/Local/R/win-library/4.2/RcppEigen/include'   -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O0 -g -Wno-ignored-attributes -c ctmc_dive.cpp -o ctmc_dive.o
```

My g++ version under x86_64-w64-mingw32 (rtools42):
```
$ g++ --version
g++.exe (GCC) 10.3.0 Copyright (C) 2020 Free Software Foundation, Inc.
```

edit: I was looking at rtools40 not 42 for a second.

**Comment by dill** [comment]
So it looks like the machine that I'm running this on has an older Rtools (only rtools40), so there might be some issue with newer builds:

```
g++.exe (Built by Jeroen for the R-project) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
```

I'll look into it but I'll need to find another Windows machine to see what's going on.

**Comment by r-glennie** [comment]
Compliation flag was issue, fixed by 234bd59
